File: CK3toEU4/Source/CK3ToEU4Converter.cpp
#include "CK3ToEU4Converter.h"
#include "Configuration/Configuration.h"
#include "CK3World/World.h"
#include "EU4World/EU4World.h"
#include "Log.h"

void convertCK3ToEU4(const mappers::ConverterVersion& converterVersion)
{
	Log(LogLevel::Progress) << "0 %";
	const auto theConfiguration = std::make_shared<Configuration>(converterVersion);
	const CK3::World sourceWorld(theConfiguration, converterVersion);
	EU4::World destWorld(sourceWorld, *theConfiguration, converterVersion);
	LOG(LogLevel::Info) << "* Conversion complete *";
	Log(LogLevel::Progress) << "100 %";
}

File: CK3ToEU4/Source/CK3ToEU4Converter.cs
using System;
using CK3ToEU4.Configuration;
using commonItems;

using CK3World = CK3ToEU4.CK3.World;

namespace CK3ToEU4;


public static class CK3ToEU4Converter
{
    public static void ConvertCK3ToEU4(ConverterVersion converterVersion)
    {
        Logger.Progress(0);
        var theConfiguration = new Config(converterVersion);
        CK3World sourceWorld = new(theConfiguration, converterVersion);
        throw new NotImplementedException();
        // TODO: EU4::World destWorld(sourceWorld, *theConfiguration, converterVersion);
        Logger.Notice("* Conversion complete *");
        Logger.Progress(100);
    }

}